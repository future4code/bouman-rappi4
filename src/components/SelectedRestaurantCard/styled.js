import styled from "styled-components";

export const CardContainer = styled.div `
    width: 90%;
    min-height: 300px;
    margin: 10px;
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: center;
    background-color: #ffffff;
    border-radius: 5px;
    :hover {
        cursor: pointer;
    }
`

export const RestaurantName = styled.div `
    width: 100%;
    height: 10px;
    color: #E86E5A;
    padding: 10px;
    font-weight: bolder;
`
export const RestaurantInfoContainer = styled.div `
    width: 100%;
    padding: 10px;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: flex-start;
    color: #bdbdbd;
    font-weight: bolder;
    margin-bottom: 8px;
`
export const ShippingAndDeliveryContainer = styled.div `
    width: 100%;
    display: flex;
    align-items: center;
    color: #bdbdbd;
    font-weight: bolder;
    margin-bottom: 8px;
`

export const StyledDiv = styled.div `
    margin-left: 30px;
`

export const CategoryDiv = styled.div `
    margin-bottom: 8px;
    margin-top: 8px;
`

export const CardImg = styled.img `
    width: 100%;
    height: 20vh;
    border-radius: 5px;
    margin-Bottom: 5px;
`