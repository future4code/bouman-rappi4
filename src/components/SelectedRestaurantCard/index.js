import React from 'react';
import { CardContainer, CardImg, RestaurantName, ShippingAndDeliveryContainer, RestaurantInfoContainer, StyledDiv, CategoryDiv } from './styled'

function SelectedRestaurantCard(props) {
    return (
        <CardContainer key={props.key} onClick={props.onClick}>
            <CardImg src={props.img} />
            <RestaurantName>{props.name}</RestaurantName>
            <RestaurantInfoContainer>
                <CategoryDiv>{props.category}</CategoryDiv>
                <ShippingAndDeliveryContainer>
                    <div>{props.deliveryTime} min </div>
                    <StyledDiv>Frete R${props.price},00</StyledDiv>
                </ShippingAndDeliveryContainer>
                <div>{props.address}</div>
            </RestaurantInfoContainer>
        </CardContainer>
    )
}

export default SelectedRestaurantCard;
