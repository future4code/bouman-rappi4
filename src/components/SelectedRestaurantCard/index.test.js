import React from 'react'
import renderer from "react-test-renderer";
import SelectedRestaurantCard from './index';

describe ('teste componente SelectedRestaurantCard', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<SelectedRestaurantCard></SelectedRestaurantCard>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
})