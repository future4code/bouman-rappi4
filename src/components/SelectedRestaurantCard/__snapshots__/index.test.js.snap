// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`teste componente SelectedRestaurantCard renders correctly 1`] = `
<div
  className="sc-bdVaJa gGKmbI"
>
  <img
    className="sc-bZQynM elEuHe"
  />
  <div
    className="sc-bwzfXH fSIllJ"
  />
  <div
    className="sc-htpNat UOjDe"
  >
    <div
      className="sc-EHOje hiLVtM"
    />
    <div
      className="sc-bxivhb fBgaFq"
    >
      <div>
         min 
      </div>
      <div
        className="sc-ifAKCX leePPw"
      >
        Frete R$
        ,00
      </div>
    </div>
    <div />
  </div>
</div>
`;
